tests/publiccloud/run_ltp.pm

```diff
@@ -59,8 +64,12 @@ sub run {
     $instance->run_ssh_command(cmd => 'sudo zypper --no-gpg-checks --gpg-auto-import-keys -q in -y ' . $remote_rpm_path, timeout => 600);
     $instance->run_ssh_command(cmd => 'sudo CREATE_ENTRIES=1 /opt/ltp/IDcheck.sh', timeout => 300);
 
-    my $reset_cmd = sprintf('~/restart_instance.sh %s %s %s %s', get_required_var('PUBLIC_CLOUD_PROVIDER'), $instance->instance_id, $instance->public_ip,
-        $provider->region);
+    my $args      = sprintf('"%s" "%s" "%s" "%s"', get_required_var('PUBLIC_CLOUD_PROVIDER'), $instance->instance_id, $instance->public_ip, $provider->region);
+    my $reset_cmd = $root_dir . '/restart_instance.sh ' . $args;
+    my $log_start_cmd = $root_dir . '/log_instance.sh start ' . $args;
+
+    assert_script_run($log_start_cmd);
+    assert_script_run(" echo '$root_dir/log_instance.sh stop $args' >> $root_dir/stop_log_instance.sh");
```

Comment:
why not just call `$root_dir/log_instance.sh stop $args` whenever you need it ?

Reply:
cause you do not have $args on cleanup.

Reply:
even that I don't like one line functions but I would really prefer it here instead complex flow which you introducing : 

```
sub generate_args {
return  sprintf('"%s" "%s" "%s" "%s"', get_required_var('PUBLIC_CLOUD_PROVIDER'), $instance->instance_id, $instance->public_ip, $provider->region);
}
```

so you can just call 
`assert_script_run("$root_dir/log_instance.sh stop " . generate_args());` whenever needed

Reply:
I see...
so I will add the $instance as class member and do as you propose.

Reply:
done